src/main/java/io/cryostat/graphql/EnvironmentNodes.java

```diff
@@ -0,0 +1,133 @@
+package io.cryostat.graphql;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.graphql.Description;
+import org.eclipse.microprofile.graphql.GraphQLApi;
+import org.eclipse.microprofile.graphql.Query;
+
+import io.cryostat.discovery.DiscoveryNode;
+import io.smallrye.graphql.api.Nullable;
+
+@GraphQLApi
+public class EnvironmentNodes {
+
+    public static class EnvironmentNodeFilterInput {
+        @Nullable
+        public Long id;
+        @Nullable
+        public String name;
+        @Nullable
+        public List<String> names;
+        @Nullable
+        public String nodeType;
+        @Nullable
+        public List<String> labels;
+
+    }
+
+    public static class EnvironmentNode {
```

Comment:
I think like in https://github.com/cryostatio/cryostat3/pull/294 , this should just be a https://github.com/cryostatio/cryostat3/blob/main/src/main/java/io/cryostat/discovery/DiscoveryNode.java .